src/main/java/com/redhat/devtools/lsp4ij/console/explorer/actions/ExportServerAction.java

```diff
@@ -0,0 +1,83 @@
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public class ExportServerAction extends AnAction {
+    private final List<LanguageServerDefinition> languageServerDefinitions;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExportServerAction.class);
+
+    public ExportServerAction(List<LanguageServerDefinition> languageServerDefinitions) {
+        this.languageServerDefinitions = languageServerDefinitions;
+        if (this.languageServerDefinitions.size() == 1) {
+            getTemplatePresentation().setText(LanguageServerBundle.message("action.lsp.console.explorer.export.server.text"));
+            getTemplatePresentation().setDescription(LanguageServerBundle.message("action.lsp.console.explorer.export.server.description"));
+        } else {
+            getTemplatePresentation().setText(LanguageServerBundle.message("action.lsp.console.explorer.export.servers.text"));
+            getTemplatePresentation().setDescription(LanguageServerBundle.message("action.lsp.console.explorer.export.servers.description"));
+        }
+        getTemplatePresentation().setIcon(AllIcons.ToolbarDecorator.Export);
+    }
+
+    @Override
+    public void actionPerformed(@NotNull AnActionEvent e) {
+        FileChooserFactory fileChooserFactory = FileChooserFactory.getInstance();
+        FileSaverDescriptor fileSaverDescriptor = new FileSaverDescriptor(
+                LanguageServerBundle.message("action.lsp.console.explorer.export.servers.zip.save.title"), LanguageServerBundle.message("action.lsp.console.explorer.export.servers.zip.save.description"));
+        FileSaverDialog fileSaverDialog = fileChooserFactory.createSaveFileDialog(fileSaverDescriptor, e.getProject());
+        VirtualFileWrapper fileWrapper = fileSaverDialog.save("export.zip");
+
+        if (fileWrapper != null) {
+            VirtualFile virtualFile = fileWrapper.getVirtualFile(true);
+            if (virtualFile != null) {
+                ApplicationManager.getApplication().runWriteAction(() -> {
+                    try {
+                        virtualFile.setBinaryContent(createZipFromStrings());
+                    } catch (IOException ex) {
+                        LOGGER.warn(ex.getLocalizedMessage(), e);
+                    }
+                });
+            }
+        }
+    }
+
+    private byte[] createZipFromStrings() throws IOException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ZipOutputStream zos = new ZipOutputStream(baos);
+
+        for (LanguageServerDefinition lsDefinition : languageServerDefinitions) {
+            String lsName = lsDefinition.getDisplayName();
+            ZipEntry entry = new ZipEntry(lsName + "/" + lsName + ".json");
+            zos.putNextEntry(entry);
+            zos.write(lsDefinition.toJsonString().getBytes(StandardCharsets.UTF_8));
```

Comment:
Instead of working with language server définition I think it should be better to create an instance of temolate https://github.com/redhat-developer/lsp4ij/blob/main/src%2Fmain%2Fjava%2Fcom%2Fredhat%2Fdevtools%2Flsp4ij%2Flaunching%2Ftemplates%2FLanguageServerTemplate.java and use gson to serialize it since it is used to load templates